src/theory/arith/nl/nonlinear_extension.cpp

```diff
@@ -557,12 +557,15 @@ int NonlinearExtension::checkLastCall(const std::vector<Node>& assertions,
   }
   if (options::nlCad())
   {
-    lemmas = d_cadSlv.checkFull();
-    if (lemmas.empty())
+    d_cadSlv.checkFull();
+    if (!d_im.hasProcessed())
```

Comment:
Should be careful here about hasPending vs. hasProcessed, this is being changed in https://github.com/CVC4/CVC4/pull/5007/files.  Can you say what you're wanting to check here?  E.g. conflict or any pending lemma (or already sent a lemma)?

Reply:
I want to check (and I guess this is the check we want to have all over the place in the nonlinear extension) whether any conflict, lemma or fact has been added. Essentially, whether the previous call has done anything to advance the solving process. Right now, this would probably be `hasSentFact() || hasSentLemma() || hasPendingFact() || hasPendingLemma() || isInConflict()`...

Reply:
Sets often uses `hasSentFact() || hasSentLemma() || isInConflict()` = `TheoryInferenceManager::hasProcessed()`.

Strings often uses `hasPendingFact() || hasPendingLemma() || isInConflict()` = `InferenceManagerBuffered::hasPending() || TheoryState::isInConflict()`.

So i think you'd want to do `TheoryInferenceManager::hasProcessed() || InferenceManagerBuffered::hasPending()` for this after the above PR is merged.  We could perhaps make this a helper method in `InferenceManagerBuffered`?

Reply:
Yes, this probably makes sense. And we should have a good naming scheme for all these functions...